Lib/test/test_call.py

```diff
@@ -457,6 +457,19 @@ def test_fastcall_keywords(self):
                 result = _testcapi.pyobject_fastcallkeywords(func, args, kwnames)
                 self.check_result(result, expected)
 
+    def test_fastcall_clearing_dict(self):
+        # Test bpo-36907: the point of the test is just checking that this
+        # does not crash.
+        class IntWithDict:
+            __slots__ = ["kwargs"]
+            def __init__(self, **kwargs):
+                self.kwargs = kwargs
+            def __index__(self):
+                self.kwargs.clear()
+                L = [2**i for i in range(10000)]
```

Comment:
This is supposed to trigger garbage collection, right? Why not make it explicit?

```suggestion
                gc.collect()
```

(with `import gc` at the top)

Reply:
> This is supposed to trigger garbage collection, right? Why not make it explicit?

Not really, it is meant to clobber memory. The bug is an use-after-free but crashing is only guaranteed if the used-after-freed memory is corrupted. But I'll add a comment explaining that.

Reply:
IMO we just leave that to the debug allocator, or valgrind runs.

Reply:
> IMO we just leave that to the debug allocator, or valgrind runs.

Fine for me, but I have to mention that a similar thing was accepted by you(!) in #9084.

Reply:
I hope the things I learned since then are useful.

@vstinner, what's your view on this? Is it useful to clobber memory with `L = [2**i for i in range(10000)]`?

Reply:
> @vstinner, what's your view on this? Is it useful to clobber memory with L = [2**i for i in range(10000)]?

I'm sorry but I don't have the bandwidth right now to follow the discussion on PEP 590 and it's implementation :-( Retry to ping me in 1 or 2 weeks if you want ;-)

Reply:
@vstinner Ah, sorry. This isn't really about PEP 590, though. It's a refcounting bug, an object is deallocated while it's still in use.

In normal builds, the bug will manifest only if you clobber the memory after it is deallocated. The debug allocator (or valgrind) should catch it even if the `L = [2**i for i in range(10000)]` is not done.

Is it OK to rely on the debug allocator (or valgrind) to prevent regressions?

(If you don't have bandwidth for that either, let me know)

Reply:
In the mean time, I removed that line.